### DISCUSSION TITLE
Default Password shema bug?

### DISCUSSION THREAD

**Discussion by fresH229a** [modoboa/modoboa#1731] [Help]
# Impacted versions

* Modoboa: 1.13.1
* installer used: Yes
* Webserver: Nginx
* Debian Version 9.9

# Steps to reproduce

![2019_05_02_20_37_32_Modoboa_settings](https://user-images.githubusercontent.com/35875328/57098369-470b2200-6d1a-11e9-852a-3ff5b5bc439b.png)

I got an error on the inital Setup that the sha256 is not a valid shema.
After reloading the page, there is only plain

**Comment by fresH229a** [comment]
If I select Information and  go back to Parameters I have this:
![2019_05_02_20_40_29_Modoboa_settings](https://user-images.githubusercontent.com/35875328/57098526-99e4d980-6d1a-11e9-829c-208a74dcdb20.png)

If I press save this:
![2019_05_02_20_41_12_Modoboa_settings](https://user-images.githubusercontent.com/35875328/57098545-a8cb8c00-6d1a-11e9-8f81-5f2f75356f4d.png)

**Comment by tonioo** [comment]
@fresH229a Are you sure your installation is complete?

**Comment by tonioo** [comment]
@fresH229a CAn you check that the modoboa system user is allowed to run the following command:
```bash
> doveadm pw -l
```

**Comment by fresH229a** [comment]
@tonioo  everything is fine on my system now since I did install those packages before installing modoboa.
No issue, Mail works fine

**Comment by DaveDischord** [comment]
> ```shell
> doveadm pw -l
> ```

What if it doesn't run correctly?

**Comment by tonioo** [comment]
@DaveDischord it means modoboa won't be able to find which algorithms are available...

**Comment by tonioo** [comment]
@fresH229a Any news?

**Comment by slimstalleren** [comment]
I have the same issue. I only have one choice: plain (weak). I would really like to use encryption. I am not sure I understand if I have to login as the modoboa user or if I can do it as a normal user of the system. 

When I execute the command, the output is
```
slimstalleren@utilityserver:~/modoboa-installer$ doveadm pw -l
MD5 MD5-CRYPT SHA SHA1 SHA256 SHA512 SMD5 SSHA SSHA256 SSHA512 PLAIN CLEAR CLEARTEXT PLAIN-TRUNC CRAM-MD5 SCRAM-SHA-1 HMAC-MD5 DIGEST-MD5 PLAIN-MD4 PLAIN-MD5 LDAP-MD5 LANMAN NTLM OTP SKEY RPA PBKDF2 CRYPT SHA256-CRYPT SHA512-CRYPT
```
So it looks like everything is available. I am running Ubuntu 18.04

**Comment by tonioo** [comment]
@phreakie Try to run this command as `modoboa` user.

**Comment by samweisgamdschie** [comment]
Just for information: we had the same issue with a fresh installation on ubuntu 20.04 and fixed it with adding the user `modoboa` to the group `dovecot` like

    sudo usermod -a -G dovecot modoboa

Because we had a `Permission denied: /var/run/dovecot/stats-writer` with the command `doveadm pw -l` 

I just wonder, is it the way to go?

**Comment by lane-ftw** [comment]
Used https://github.com/modoboa/modoboa-installer
Ubuntu 22.04 unprivileged LXC on proxmox 8.1.3 
Having a similar issue, only I have plain and md5 as options. I do not get any errors when running the installer.
Was getting `Permission denied: /var/run/dovecot/stats-writer` until I added modoboa to vmail group.
`usermod -a -G vmail modoboa`
Now modoba seems to have permission, but I still only have plain and md5.
```
root@mail:~# su modoboa
$ doveadm pw -l
SHA1 SSHA512 SCRAM-SHA-256 BLF-CRYPT PLAIN HMAC-MD5 OTP SHA512 SHA DES-CRYPT CRYPT SSHA MD5-CRYPT PLAIN-MD4 PLAIN-MD5 SCRAM-SHA-1 SHA512-CRYPT CLEAR CLEARTEXT ARGON2I ARGON2ID SSHA256 MD5 PBKDF2 SHA256 CRAM-MD5 PLAIN-TRUNC SHA256-CRYPT SMD5 DIGEST-MD5 LDAP-MD5
```

EDIT: Tried on a privileged container, no change, made sure there was no mention of noexec on ANY tmp filesystem, no change.
What logs should I be looking at to try and diagnose this?

EDIT2: Thanks to Spitap for the fix!
Steps were:
```
sudo -u modoboa -i bash
/srv/modoboa/env/bin/python /srv/modoboa/instance/manage.py shell
from django.core.cache import cache
cache.get('password_scheme_choice')
cache.delete("password_scheme_choice")
```
Fix incoming I'm told.
